### PR TITLE
*: remove the support of the old ddl framework

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/39824

remove the support of the old ddl framework

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -408,10 +408,6 @@ bazel_coverage_test: failpoint-enable bazel_ci_prepare
 		--build_event_json_file=bazel_1.json --@io_bazel_rules_go//go/config:cover_format=go_cover \
 		-- //... -//cmd/... -//tests/graceshutdown/... \
 		-//tests/globalkilltest/... -//tests/readonlytest/... -//br/pkg/task:task_test -//tests/realtikvtest/...
-	bazel $(BAZEL_GLOBAL_CONFIG) coverage $(BAZEL_CMD_CONFIG) \
```
Comment (Defined2014) [inline Makefile:411]: featuretag will also used by other feature. Do not delete it.

Comment (wjhuang2016) [reply Makefile:411]: We should not enable it only for distributing reorg, so I remove it now.

**Review by hawkingrei — APPROVED** [review]

File: meta/meta.go [hunk 2]
```diff
@@ -77,7 +74,6 @@ var (
 	mPolicyGlobalID   = []byte("PolicyGlobalID")
 	mPolicyMagicByte  = CurrentMagicByteVer
 	mDDLTableVersion  = []byte("DDLTableVersion")
-	mConcurrentDDL    = []byte("concurrentDDL")
```
Comment (Defined2014) [inline meta/meta.go:80]: Do we need to remove the corresponding value stored in the TiKV cluster when upgrading? Although the space used is not much, it is better not to clean it up.

Comment (wjhuang2016) [reply meta/meta.go:80]: We can just leave it here. Less code, less bug :)

File: br/pkg/version/version.go [hunk 3]
```diff
@@ -166,9 +165,7 @@ func CheckVersionForDDL(s *metapb.Store, tikvVersion *semver.Version) error {
 	// use tikvVersion instead of tidbVersion since br doesn't have mysql client to connect tidb.
 	requireVersion := semver.New("6.2.0-alpha")
 	if tikvVersion.Compare(*requireVersion) < 0 {
-		log.Info("detected the old version of tidb cluster. set enable concurrent ddl to false")
-		variable.EnableConcurrentDDL.Store(false)
-		return nil
+		return errors.New("detected the old version of tidb cluster")
```
Comment (tangenta) [inline br/pkg/version/version.go:168]: Please refine the error message like "TiKV version must be greater than 6.2.0, but got X".

File: ddl/ddl_test.go [hunk 4]
```diff
@@ -272,6 +272,7 @@ func TestBuildJobDependence(t *testing.T) {
 }
 
 func TestNotifyDDLJob(t *testing.T) {
+	t.Skip("skip")
```
Comment (tangenta) [inline ddl/ddl_test.go:275]: Please explain why is this skipped.

File: ddl/ddl_worker.go [hunk 5]
```diff
@@ -295,7 +208,7 @@ func (d *ddl) addBatchDDLJobs(tasks []*limitJobTask) {
 	startTime := time.Now()
 	var err error
 	// DDLForce2Queue is a flag to tell DDL worker to always push the job to the DDL queue.
-	toTable := variable.EnableConcurrentDDL.Load() && !variable.DDLForce2Queue.Load()
+	toTable := !variable.DDLForce2Queue.Load()
```
Comment (tangenta) [inline ddl/ddl_worker.go:211]: Is the variable `DDLForce2Queue` still useful?

Comment (wjhuang2016) [reply ddl/ddl_worker.go:211]: May be used when upgrading from old cluster.

File: ddl/ddl_worker.go [hunk 6]
```diff
@@ -482,13 +419,7 @@ func (w *worker) updateDDLJob(t *meta.Meta, job *model.Job, meetErr bool) error
...
 			zap.String("job", job.String()))
 	}
-	var err error
-	if w.concurrentDDL {
-		err = updateDDLJob2Table(w.sess, job, updateRawArgs)
-	} else {
-		err = t.UpdateDDLJob(0, job, updateRawArgs)
-	}
-	return errors.Trace(err)
+	return errors.Trace(updateDDLJob2Table(w.sess, job, updateRawArgs))
```
Comment (tangenta) [inline ddl/ddl_worker.go:422]: Please remove the function parameter `t *meta.Meta` as well.

File: ddl/ddl_worker.go [hunk 7]
```diff
@@ -716,7 +643,7 @@ func isDependencyJobDone(t *meta.Meta, job *model.Job) (bool, error) {
 
 func newMetaWithQueueTp(txn kv.Transaction, tp workerType) *meta.Meta {
 	if tp == addIdxWorker {
-		return meta.NewMeta(txn, meta.AddIndexJobListKey)
+		return meta.NewMeta(txn)
```
Comment (tangenta) [inline ddl/ddl_worker.go:646]: This function is unused. Please consider removing it.

File: ddl/schema_test.go [hunk 8]
```diff
@@ -164,14 +164,7 @@ func testDropSchema(t *testing.T, ctx sessionctx.Context, d ddl.DDL, dbInfo *mod
...
 func isDDLJobDone(test *testing.T, t *meta.Meta) bool {
-	job, err := t.GetDDLJobByIdx(0)
-	require.NoError(test, err)
-	if job == nil {
-		return true
-	}
-
-	time.Sleep(testLease)
-	return false
+	return true
```
Comment (tangenta) [inline ddl/schema_test.go:167]: The DDL job is always done??

File: ddl/ddl_worker.go (hunk 5 above)
Comment (tangenta) [reply ddl/ddl_worker.go:211]: Can't we make the assumption like "when the cluster is upgrading to this version, the concurrent DDL must be enabled"?

File: ddl/ddl.go [hunk 9]
```diff
@@ -799,9 +775,6 @@ func (d *ddl) close() {
 		d.generalDDLWorkerPool.close()
 	}
 
-	for _, worker := range d.workers {
-		worker.Close()
-	}
```
Comment (tangenta) [inline ddl/ddl.go:804]: Why is this removed?

File: meta/meta.go [hunk 10]
```diff
@@ -129,17 +125,13 @@ type Meta struct {
...
 	txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
 	t := structure.NewStructure(txn, txn, mMetaPrefix)
-	listKey := DefaultJobListKey
-	if len(jobListKeys) != 0 {
-		listKey = jobListKeys[0]
-	}
 	return &Meta{txn: t,
 		StartTS:    txn.StartTS(),
-		jobListKey: listKey,
+		jobListKey: DefaultJobListKey,
```
Comment (tangenta) [inline meta/meta.go:134]: The `jobListKey` is not used anymore. Please remove it.

**Review by tangenta — COMMENTED** [review]
Several tests are removed because the implementation is "DDL queue". I am not sure if we should rewrite these tests for "DDL table" version.

File: ddl/ddl.go (hunk 9 above)
Comment (wjhuang2016) [reply ddl/ddl.go:804]: No used anymore

File: br/pkg/version/version.go [hunk 11]
```diff
@@ -166,9 +165,7 @@ func CheckVersionForDDL(s *metapb.Store, tikvVersion *semver.Version) error {
 	// use tikvVersion instead of tidbVersion since br doesn't have mysql client to connect tidb.
 	requireVersion := semver.New("6.2.0-alpha")
 	if tikvVersion.Compare(*requireVersion) < 0 {
-		log.Info("detected the old version of tidb cluster. set enable concurrent ddl to false")
-		variable.EnableConcurrentDDL.Store(false)
-		return nil
+		return errors.Errorf("detected the old version of tidb cluster, require: >= 6.2.0, but got %s", tikvVersion.String())
```
Comment (Benjamin2037) [inline br/pkg/version/version.go:168]:
```suggestion
		return errors.Errorf("detected the old version of tidb cluster, require: >= 6.2.0, but got %s", tidbVersion.String())？
```

**Review by tangenta — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (bb7133) [reply Makefile:411]: Do u mean that we can add it back when it is used by other features?

**Review by bb7133 — APPROVED** [review]
LGTM
